Sources/_StringProcessing/Engine/Registers.swift

```diff
@@ -156,6 +158,22 @@ extension Processor.Registers {
 
     self.savePointAddresses = Array(repeating: 0, count: info.savePointAddresses)
   }
+
+  mutating func reset(_ sentinel: Input.Index) {
+    // note: Is there any issue with the program transform functions holding
+    // state and not getting reset here? Do we care?
+    self.bools = Array(repeating: false, count: info.bools)
+    self.ints = Array(repeating: 0, count: info.ints)
+    self.floats = Array(repeating: 0, count: info.floats)
+    self.positions = Array(repeating: sentinel, count: info.positions)
+    self.values = Array(
+      repeating: SentinelValue(), count: info.values)
+
+    self.instructionAddresses = Array(repeating: 0, count: info.instructionAddresses)
+    self.classStackAddresses = Array(repeating: 0, count: info.classStackAddresses)
+    self.positionStackAddresses = Array(repeating: 0, count: info.positionStackAddresses)
+    self.savePointAddresses = Array(repeating: 0, count: info.savePointAddresses)
+  }
```

Comment:
Just clear these instead of allocating new arrays. Making new arrays is what makes creating a new processor slow.